### ISSUE TITLE
Look into having dedicated nodes for core kubernetes sevices and not allowing user pods on those nodes

### ISSUE THREAD

**Issue by moorepants** [LibreTexts/metalc#106]

**Comment by lux12337** [comment]
Looks like we need to setup affinity on the nodes and set the [scheduling.corePods](https://zero-to-jupyterhub.readthedocs.io/en/stable/reference.html#singleuser-schedulerstrategy)
![image](https://user-images.githubusercontent.com/19236515/70680088-ed71bb00-1c4b-11ea-8d46-f975795c180d.png)

**Comment by rkevin-arch** [comment]
What is the purpose of this? If by "core kubernetes services" you mean stuff like the kube API server, CoreDNS, etcd and friends, then this is already done by kubernetes by default. They will only get scheduled on the control plane node chick0, and no other pods will spawn on chick0. If you mean separating the hub/proxy pods from the user pods, we can do that by using node labels like Luigi posted. Let me know if this is something worth spending some time on.

**Comment by moorepants** [comment]
We wrote this because we were having lots of crashes and dying user pods that would lock up a node (if I remember correctly). We had discussed preventing user session pods from being created on nodes that had critical services running on them to prevent critical services from being lost. But this is probably irrelevant now, as things seems to be running smoothly (knock on wood). @rkevin-arch if you agree, then close this.
